src/borg/repository.py

```diff
@@ -353,7 +357,7 @@ def write_index(self):
     def check_free_space(self):
         """Pre-commit check for sufficient free space to actually perform the commit."""
         required_free_space = self.index.size_on_disk() * 4
-        # TODO: hints file size
+        required_free_space += len(msgpack.packb(self.hints()))
```

Comment:
probably better to just make a conservative estimate. this copies the entire structure just to get the length.